source/schema.d.ts

```diff
@@ -48,7 +50,9 @@ export type Schema<ObjectType, ValueType> = ObjectType extends string
 				: ObjectType extends ReadonlySet<unknown>
 					? ValueType
 					: ObjectType extends Array<infer U>
-						? Array<Schema<U, ValueType>>
+						? Options['recurseIntoArrays'] extends false
```

Comment:
Question: Because `recurseIntoArrays` is optional, shouldn't we check here also for undefined? 🤔 

```suggestion
						? Options['recurseIntoArrays'] extends false | undefined
```

Reply:
If the user has TS Enabled `exactOptionalPropertyTypes`, this should not be a problem.
If the user does not have this enabled, they can pass undefined for recurseIntoArrays

Reply:
Probably we also need a test case for when `undefined` is passed. Because I see that type-fest has enabled `exactOptionalPropertyTypes`

Reply:
I included the `undefined` in the tests when using `recurseIntoArrays`